lib/segment/src/lib.rs

```diff
@@ -20,6 +20,21 @@ pub mod vector_storage;
 extern crate num_derive;
 extern crate core;
 
+static MMAP_ADVISE: parking_lot::RwLock<memmap2::Advice> =
```

Comment:
But I hope we will remove static value in future =)

Reply:
> `memmap2::Advice` is just i32 value. I propose to use atomics instead of RwLock

Nice! Haven't noticed `#[repr(i32)]` initially. I we could use [`atomic_cell`] for this.

> But I hope we will remove static value in future =)

I'm not a fan of a static too, but it seems like a "proper" implementation would require an excruciating amount of refactoring. :/

[`atomic_cell`]: https://docs.rs/atomic_cell/0.2.0/atomic_cell/

Reply:
It will still look good inthe code, right? We'll still be using enums, and not weird constants?

Reply:
With `atomic_cell` - sure. It's, basically, hides the "transmutation" into/from underlying `i32`.